app/src/androidTest/java/fr/free/nrw/commons/UploadActivityTest.kt

```diff
@@ -12,8 +21,31 @@ class UploadActivityTest {
     @get:Rule
     var activityRule = ActivityTestRule(UploadActivity::class.java)
 
-    @Test
+ @Test
     fun orientationChange() {
         UITestHelper.changeOrientation(activityRule)
     }
-}
\ No newline at end of file
+
+    @Test
+    fun TestForCaptionsAndDepictions() {
+        val imageUri = Uri.parse("file://mnt/sdcard/image.jpg")
+
+        Espresso.onView(ViewMatchers.withId(R.id.caption_item_edit_text))
+                .perform(ViewActions.typeText("caption in english"))
+        Espresso.onView(ViewMatchers.withId(R.id.description_item_edit_text))
+                .perform(ViewActions.typeText("description in english"))
+        Espresso.onView(ViewMatchers.withId(R.id.spinner_description_languages))
+                .perform(ViewActions.click())
+        Espresso.onView(ViewMatchers.withId(R.id.spinner_description_languages)).perform(ViewActions.click());
+        Espresso.onData(AllOf.allOf(Matchers.anything("spinner text"))).atPosition(1).perform(ViewActions.click());
+        Espresso.onView(ViewMatchers.withId(R.id.caption_item_edit_text))
+                .perform(ViewActions.typeText("caption in some other language"))
+        Espresso.onView(ViewMatchers.withId(R.id.description_item_edit_text))
+                .perform(ViewActions.typeText("description in some other language"))
+        Espresso.onView(ViewMatchers.withId(R.id.btn_next))
```

Comment:
Test for captions and Depictions is in two files ?

Reply:
This is for the base test mentioned here https://github.com/commons-app/apps-android-commons/issues/3031. I thoought it should all be in a continuation that's why I have put it in a single class